flatfs_test.go

```diff
@@ -400,6 +401,50 @@ func testBatchDelete(dirFunc mkShardFunc, t *testing.T) {
 
 func TestBatchDelete(t *testing.T) { tryAllShardFuncs(t, testBatchDelete) }
 
+func TestNoCluster(t *testing.T) {
+	tempdir, cleanup := tempdir(t)
+	defer cleanup()
+
+	fs, err := flatfs.New(tempdir, flatfs.NextToLast(1), false)
+	if err != nil {
+		t.Fatalf("New fail: %v\n", err)
+	}
+
+	r := rand.New()
+	N := 1024 // should be divisible by 32 so the math works out
+	for i := 0; i < N; i++ {
+		blk := make([]byte, 1000)
+		r.Read(blk)
+
+		key := "CIQ" + base32.StdEncoding.EncodeToString(blk[:10])
+		err := fs.Put(datastore.NewKey(key), blk)
+		if err != nil {
+			t.Fatalf("Put fail: %v\n", err)
+		}
+	}
+
+	dirs, err := ioutil.ReadDir(tempdir)
+	if err != nil {
+		t.Fatalf("ReadDir fail: %v\n", err)
+	}
+	if len(dirs) != 32 {
+		t.Fatalf("Expected 32 directories in %s", tempdir)
+	}
+	idealFilesPerDir := float64(N) / 32.0
+	tolerance := math.Floor(idealFilesPerDir * 0.50)
```

Comment:
That is 50% tolerance, right? Can we still reliably get closer than this?

We might also make the test bigger but make the seed static, as currently there is remote but non-zero chance of this test failing if stars align correctly.